src/metabase/api/user.clj

```diff
@@ -78,7 +78,8 @@
   {password     [Required ComplexPassword]}
   (check-self-or-superuser id)
   (let-404 [user (db/select-one [User :password_salt :password], :id id, :is_active true)]
-    (when (= id *current-user-id*)
+    (when (and (not (:is_superuser @*current-user*))
+               (= id *current-user-id*))
```

Comment:
it probably doesn't make sense to have the `check-self-or-superuser` call and then a few lines later a second call to check that the current user *isn't* a superuser